submarine-server/server-api/src/main/java/org/apache/submarine/server/api/spec/EnvironmentSpec.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.server.api.spec;
+
+/**
+ * Submarine environment spec.
+ */
+/**
+ * @author mani
+ *
+ */
+public class EnvironmentSpec {
+
+  
+  /**
+   * Name of the environment
+   */
+  private String name;
+
+  /**
+   * Docker-Image of the environment
+   */
+  private String dockerImage;
+  
+  /**
+   * Kernel Spec
+   */
+  private KernelSpec kernelSpec;
+  
+  /**
+   * Description
+   */
+  private String description;
+
+  /**
+   * Image
+   * @return
+   */
+  private String image;
+  
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public String getDockerImage() {
+    return dockerImage;
+  }
+
+  public void setDockerImage(String dockerImage) {
+    this.dockerImage = dockerImage;
+  }
+
+  public KernelSpec getKernelSpec() {
+    return kernelSpec;
+  }
+
+  public void setKernelSpec(KernelSpec kernelSpec) {
+    this.kernelSpec = kernelSpec;
+  }
+
+  /**
+   * @return the description
+   */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * @param description the description to set
+   */
+  public void setDescription(String description) {
+    this.description = description;
+  }
+
+  /**
+   * @return the image
+   */
+  public String getImage() {
+    return image;
+  }
```

Comment:
I guess this is for VM image, can we rename it to vmimage?

Reply:
Had vmimage in the very beginning. Then removed it after your comments mentioned on https://issues.apache.org/jira/browse/SUBMARINE-507?focusedCommentId=17116299&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17116299.

Now, it had come again because of recent commit - to handle below environment definition

environment: {
"image"...
}

of ExperimentSpec in EnvironmentSpec itself. In fact, raised a question about the same also. Please refer https://github.com/apache/submarine/pull/299#issuecomment-648614756.

Reply:
I see, then I suggest to rename it to "containerImage" to make it more specific. "Image" maybe too general.

Reply:
Experiment Spec contains "image" and we are about to ship 0.4.0 release. So, renaming as "containerImage" cause unnecessary confusions in upcoming releases. If we want to rename, it is better to do for 0.4.0 release itself. Thoughts?

Reply:
@manirajv06 , sorry I missed your last comment, we now have both dockerImage and image, I think we should only leave dockerImage. 

From 0.4.0 to 0.5.0 it is fine to break compatibility, since 0.4.0 we explicitly called out that it is an alpha release. Other experiment API will be unavoidably updated too (in an incompatible way).

Reply:
Slightly changed the definition of "environment" in Experiment spec. For example, Below spec is valid.

environment: {
"image":"..",
"name":".."
}

If "image" is available, set it as k8 container image. If "name" is available, fetch the environment details using name from DB/Cache and use the details for K8 init container process. Hope this make the whole thing intuitive and simpler. Thoughts? @tangzhankun @wangdatan